### PR TITLE
Rhmap 3085 fix download file

### PR DESCRIPTION
**Pull request by wei-lee**

### CODE REVIEW

**Comment by fheng-build-servers** [comment]
@jasonmadigan @david-martin please review this PR.

**Comment by wei-lee** [comment]
@mmurphy  can you review this please? I cherry-picked your fix and add some extra code to better handling the errors.

File: lib/utils/request.js [hunk 1]
```diff
@@ -213,14 +215,22 @@ function downloadFile(params, cb){
...
-    return cb(err);
+    fs.unlink(filePath, function(){
+      return cbOnce(err);
+    });
   });
 
-  req.on('end', function(){
-    return cb();
+  fileStream.on('finish', function(){
+    console.log('Saved to ' + filePath);
```
Comment (mmurphy) [inline lib/utils/request.js]: should this be a logger call?

Comment (wei-lee) [inline lib/utils/request.js]: I tried to use the log.info, but it doesn't output anything. The log in fhc is soooo complicated...

File: npm-shrinkwrap.json [hunk 2]
```diff
@@ -202,9 +190,219 @@
...
         "inherits": {
-          "version": "1.0.1",
+          "version": "1.0.2",
           "from": "inherits@~1.0.0",
-          "resolved": "https://registry.npmjs.org/inherits/-/inherits-1.0.1.tgz"
+          "resolved": "https://registry.npmjs.org/inherits/-/inherits-1.0.2.tgz"
+        }
+      }
+    },
+    "grunt": {
```
Comment (mmurphy) [inline npm-shrinkwrap.json]: should grunt be in the shrinkwrap?

Comment (wei-lee) [inline npm-shrinkwrap.json]: yeah, probably not. Let me regenerate it. such a PITA

File: lib/utils/request.js (hunk 1 above)
Comment (mmurphy) [inline lib/utils/request.js]: the following should work:
log.info('Saved to ' + filePath,'file download');

**Comment by mmurphy** [comment]
looks fine

**Pull request merged**